### PR TITLE
Updating .NET 7 support to RC2, adding support for Framework 4.8.1

### PR DESCRIPTION
**Pull request by chynesNR**
* Upgraded from .NET 7 RC1 to RC2
* Filled in some missing test coverage
* Added test support for Framework 4.8.1

### CODE REVIEW

File: src/NewRelic.Core/DotnetVersion.cs [hunk 1]
```diff
@@ -29,6 +29,7 @@ public enum DotnetCoreVersion
         netcoreapp31,
         net5,
         net6,
+        net7,
```
Comment (JcolemanNR) [inline src/NewRelic.Core/DotnetVersion.cs:32]: Needs detection for 4.8.1

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj [hunk 2]
```diff
@@ -180,21 +211,30 @@
     <PackageReference Include="Microsoft.Extensions.Logging.Configuration" Version="6.0.0" Condition="'$(TargetFramework)' == 'net6.0'" />
     <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="6.0.0" Condition="'$(TargetFramework)' == 'net6.0'" />
     <PackageReference Include="Serilog.Extensions.Hosting" Version="4.2.0" Condition="'$(TargetFramework)' == 'net6.0'" />
+
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="7.0.0-preview.6.22324.4" Condition="'$(TargetFramework)' == 'net7.0'" />
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj:215]: Hmm... not sure about the preview 6 reference here

File: tests/Agent/IntegrationTests/IntegrationTests/HttpClientInstrumentation/HttpClientInstrumentation.cs [hunk 3]
```diff
@@ -163,9 +172,9 @@ public HttpClientInstrumentationTests_FW471(ConsoleDynamicMethodFixtureFW471 fix
     }
 
     [NetFrameworkTest]
-    public class HttpClientInstrumentationTests_FW48 : HttpClientInstrumentationTestsBase<ConsoleDynamicMethodFixtureFW48>
+    public class HttpClientInstrumentationTests_FW481 : HttpClientInstrumentationTestsBase<ConsoleDynamicMethodFixtureFW481>
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/IntegrationTests/HttpClientInstrumentation/HttpClientInstrumentation.cs:175]: revert this, and add a new test target for `ConsoleDynamicMethodFixtureFWLatest`

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
A few requested changes, and an open question

File: src/NewRelic.Core/DotnetVersion.cs (hunk 1 above)
Comment (chynesNR) [reply src/NewRelic.Core/DotnetVersion.cs:32]: Good catch, thanks

File: tests/Agent/IntegrationTests/IntegrationTests/HttpClientInstrumentation/HttpClientInstrumentation.cs (hunk 3 above)
Comment (chynesNR) [reply tests/Agent/IntegrationTests/IntegrationTests/HttpClientInstrumentation/HttpClientInstrumentation.cs:175]: Will do

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj (hunk 2 above)
Comment (chynesNR) [reply tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj:215]: I'm not sure what that's about either. I'll replace it.

**Review by vuqtran88 — APPROVED** [review]
Looks good to me. 👍

File: .github/workflows/all_solutions.yml [hunk 4]
```diff
@@ -680,6 +702,12 @@ jobs:
           dotnet-version: '6.0.x'
```
Comment (JcolemanNR) [inline .github/workflows/all_solutions.yml:702]: .net 6 is already on the runner image

File: .github/workflows/all_solutions.yml [hunk 5]
```diff
@@ -398,6 +398,16 @@ jobs:
...
```
[the remark this answers is not in the document]
Comment (chynesNR) [reply .github/workflows/all_solutions.yml]: Good idea. Time for some cleanup.

**Review by JcolemanNR — APPROVED** [review]

**Pull request merged**